v2/streaming-data-generator/src/test/java/com/google/cloud/teleport/v2/templates/StreamingDataGeneratorIT.java

```diff
@@ -0,0 +1,138 @@
+import com.google.cloud.storage.Storage;
+import com.google.cloud.teleport.it.TestProperties;
+import com.google.cloud.teleport.it.artifacts.Artifact;
+import com.google.cloud.teleport.it.artifacts.ArtifactClient;
+import com.google.cloud.teleport.it.artifacts.GcsArtifactClient;
+import com.google.cloud.teleport.it.dataflow.DataflowOperator;
+import com.google.cloud.teleport.it.dataflow.DataflowOperator.Result;
+import com.google.cloud.teleport.it.dataflow.DataflowTemplateClient;
+import com.google.cloud.teleport.it.dataflow.DataflowTemplateClient.JobInfo;
+import com.google.cloud.teleport.it.dataflow.DataflowTemplateClient.JobState;
+import com.google.cloud.teleport.it.dataflow.DataflowTemplateClient.LaunchConfig;
+import com.google.cloud.teleport.it.dataflow.FlexTemplateClient;
+import com.google.cloud.teleport.v2.templates.StreamingDataGenerator.SinkType;
+import com.google.common.io.Resources;
+import com.google.re2j.Pattern;
+import java.io.IOException;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Integration test for {@link StreamingDataGenerator}. */
+@RunWith(JUnit4.class)
+public final class StreamingDataGeneratorIT {
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final Credentials CREDENTIALS = TestProperties.googleCredentials();
+  private static final String PROJECT = TestProperties.project();
+  private static final String REGION = TestProperties.region();
+  private static final String SPEC_PATH = TestProperties.specPath();
+
+  private static final String SCHEMA_FILE = "gameevent.json";
+  private static final String LOCAL_SCHEMA_PATH = Resources.getResource(SCHEMA_FILE).getPath();
+
+  private static final String TEST_ROOT_DIR = "streaming-data-generator";
+
+  private static final String NUM_SHARDS_KEY = "numShards";
+  private static final String OUTPUT_DIRECTORY_KEY = "outputDirectory";
+  private static final String QPS_KEY = "qps";
+  private static final String SCHEMA_LOCATION_KEY = "schemaLocation";
+  private static final String SINK_TYPE_KEY = "sinkType";
+  private static final String WINDOW_DURATION_KEY = "windowDuration";
+
+  private static final String DEFAULT_QPS = "15";
+  private static final String DEFAULT_WINDOW_DURATION = "60s";
+
+  private static ArtifactClient artifactClient;
+
+  @BeforeClass
+  public static void setUpClass() throws IOException {
+    Storage gcsClient = createGcsClient(CREDENTIALS);
+    artifactClient = GcsArtifactClient.builder(gcsClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+    artifactClient.uploadArtifact(SCHEMA_FILE, LOCAL_SCHEMA_PATH);
+  }
+
+  @AfterClass
+  public static void tearDownClass() {
+    artifactClient.cleanupRun();
+  }
+
+  @Test
+  public void testFakeMessagesToGcs() throws IOException {
+    // Arrange
+    String name = "teleport-flex-streaming-data-generator-gcs";
```

Comment:
Same. Auto-generated based on the test name? (Available through https://github.com/junit-team/junit4/wiki/Rules#testname-rule).

Reply:
This does have a minor drawback in that method names aren't compatible with Dataflow job names, but that's easy enough to account for.